applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -0,0 +1,240 @@
+            process.ExecuteBeforeSolutionLoop()
+            
+        if self.have_output:
+            self.output.ExecuteBeforeSolutionLoop()
+
+
+    def _SetUpGiDOutput(self):
+        '''Initialize self.output as a GiD output instance.'''
+        self.have_output = self.project_parameters.Has("output_configuration")
+        if self.have_output:
+            if self.parallel_type == "OpenMP":
+                from gid_output_process import GiDOutputProcess as OutputProcess
+            elif self.parallel_type == "MPI":
+                from gid_output_process_mpi import GiDOutputProcessMPI as OutputProcess
+
+            self.output = OutputProcess(self.model_part,
+                                        self.project_parameters["problem_data"]["problem_name"].GetString() ,
+                                        self.project_parameters["output_configuration"])
+
+            self.output.ExecuteInitialize()
+
+    def _SetUpRestart(self):
+        """Initialize self.restart_utility as a RestartUtility instance and check if we need to initialize the problem from a restart file."""
+        if self.project_parameters.Has("restart_settings"):
+            restart_settings = self.project_parameters["restart_settings"]
+            self.load_restart = restart_settings["load_restart"].GetBool()
+            self.save_restart = restart_settings["save_restart"].GetBool()
+            restart_settings.RemoveValue("load_restart")
+            restart_settings.RemoveValue("save_restart")
+            restart_settings.AddValue("input_filename", self.project_parameters["problem_data"]["problem_name"])
+            restart_settings.AddValue("echo_level", self.project_parameters["problem_data"]["echo_level"])
+
+            if self.parallel_type == "OpenMP":
+                from restart_utility import RestartUtility as Restart
+            elif self.parallel_type == "MPI":
+                from trilinos_restart_utility import TrilinosRestartUtility as Restart
+
+            self.restart_utility = Restart(self.main_model_part,
+                                           self.project_parameters["restart_settings"])
+        else:
+            self.load_restart = False
+            self.save_restart = False
+
+    def RunMainTemporalLoop(self):
+        '''The main solution loop.'''
+        while self.time <= self.end_time:
+            self.InitializeSolutionStep()
+            self.SolveSingleStep()
+            self.FinalizeSolutionStep()
+
+    def InitializeSolutionStep(self):
+        dt = self.solver.ComputeDeltaTime()
+        self.time = self.time + dt
+        self.step = self.step + 1
+        
+        self.main_model_part.CloneTimeStep(self.time)
+        self.main_model_part.ProcessInfo[STEP] = self.step
+
+        if self.is_printing_rank:
+            Logger.PrintInfo("Fluid Dynamics Analysis","STEP = ", self.step)
+            Logger.PrintInfo("Fluid Dynamics Analysis","TIME = ", self.time)
+        
```

Comment:
I suggest to flush stdout `sys.stdout.flush()` after printing the time (on all ranks, not just the printing one)

We had some issues with this in the past

Reply:
Hi @philbucher I'm not a fan of flushing: in clusters it typically implies a disk access, which can quickly become a bottleneck for large simulations (think a hundred threads accessing the hard disk at the same time). I'll consider it for the openmp side, though.

Reply:
Hm I see
Do you have a better suggestion for doing this? Especially for debugging it is quite a pain if e.g. you don't know if an operation was done but hasn't flushed or it wasn't executed

Reply:
@philbucher i agree with @jcotela on the flushing. The nice thing of using the Logger is that everything will go through the same buffer, so it should avoid unwanted changes in order

Reply:
I would go for manipulating the output class for the logger, but I haven't really worked out the details yet. My idea would be to redirect the logger messages through an output class "optimized"  for shell ouptut in OpenMP runs (with as many flushes as you wish) or to a different one, written with file output in mind, for MPI runs. The goal, in any case, would be not to put the flushes in the common parts of the code.

Reply:
ok, thx for the explanation